File: C/4-mad-lib.c
#include <stdio.h>
#include <string.h>

int main(int argc, char *argv[]) {
    char noun[32], verb[32], adjective[32], adverb[32];

    printf("Enter a noun: ");
    fgets(noun, 32, stdin);

    printf("Enter a verb: ");
    fgets(verb, 32, stdin);

    printf("Enter a adjective: ");
    fgets(adjective, 32, stdin);

    printf("Enter a adverb: ");
    fgets(adverb, 32, stdin);

    noun[strcspn(noun, "\n")] = 0;
    verb[strcspn(verb, "\n")] = 0;
    adjective[strcspn(adjective, "\n")] = 0;
    adverb[strcspn(adverb, "\n")] = 0;

    printf("Do you %s your %s %s %s? That's hilarious!\n", verb, adjective, noun, adverb);
    
    return 0;
}

File: go/4-mad-lib.go
package main

import "fmt"

func main() {
	var noun, verb, adjective, adverb string

	fmt.Printf("Enter a noun: ")
	fmt.Scan(&noun)
	fmt.Printf("Enter a verb: ")
	fmt.Scan(&verb)
	fmt.Printf("Enter an adjective: ")
	fmt.Scan(&adjective)
	fmt.Printf("Enter an adverb: ")
	fmt.Scan(&adverb)

	fmt.Printf("Do you %s your %s %s %s? That's hilarious!\n", verb, adjective, noun, adverb)
}